Foundation/FileManager.swift

```diff
@@ -503,16 +503,16 @@ open class FileManager : NSObject {
         let buffer = UnsafeMutablePointer<UInt8>.allocate(capacity: Int(fileInfo.st_blksize))
         defer { buffer.deallocate() }
 
-        var bytesRemaining = Int64(fileInfo.st_size)
+        var bytesRemaining = Int(fileInfo.st_size)
         while bytesRemaining > 0 {
```

Comment:
`st_size` might be 64bits even on 32bit platforms, so this avoids any problems by always converting upto 64bits. The rest of the casts in this section are so that all arithmetic is conducted in 64bits

Reply:
Oh I didn't consider this. So is the cast down from `Int64` to `Int` in the following line safe on 32bit platforms because fileInfo.st_blksize is Int32 on all platforms? 

```swift
let bytesRead = try _readFrom(fd: srcfd, toBuffer: buffer, length: Int(bytesToRead), filename: srcPath)
```

Reply:
Also would is this just implicit knowledge someone should know or would it be prudent to include in a comment?

Reply:
Yes, its unlikely `st_blksize` will be 64bit on any platform and even if it was, its value would be unlikely to exceed `Int32.max`. If it ever is, it will trap and abort the program and can be dealt with then.

`bytesRemaining` is computed in 64bits as file size can be large, hence `bytesToRead` also needs to be computed in 64bits. However the blocks that are read/written will be a lot smaller and `read()` takes an `int` hence the cast.